lambdas/config.py

```diff
@@ -1,115 +1,147 @@
+            f"{logging.getLevelName(root_logger.getEffectiveLevel())}"
         )
-        raise ValueError(message)
-
-    # If next step is extract step, required harvest fields are present
-    # ruff: noqa: SIM102
-    if input_data["next-step"] == "extract":
-        if input_data["source"] not in GIS_SOURCES:
-            if missing_harvest_fields := [
-                field for field in REQUIRED_OAI_HARVEST_FIELDS if field not in input_data
-            ]:
-                message = (
-                    "Input must include all required harvest fields when starting with "
-                    f"harvest step. Missing fields: {missing_harvest_fields}"
-                )
-                raise ValueError(message)
-
-
-def verify_env() -> None:
-    """Confirm that required env variables are set."""
-    for key in REQUIRED_ENV:
-        if not os.getenv(key):
-            message = f"Required env variable {key} is not set"
-            raise RuntimeError(message)
+
+    @classmethod
+    def validate_input(cls, input_data: dict) -> None:
+        """Validate input to the lambda function.
+
+        Ensures that all requiered input fields are present and contain valid data.
+        """
+        # All required fields are present
+        if missing_fields := [
+            field for field in cls.REQUIRED_FIELDS if field not in input_data
+        ]:
+            message = (
+                "Input must include all required fields. "
+                f"Missing fields: {missing_fields}"
+            )
+            raise ValueError(message)
+
+        # Valid next step
+        next_step = input_data["next-step"]
+        if next_step not in cls.VALID_STEPS:
+            message = (
+                f"Input 'next-step' value must be one of: {cls.VALID_STEPS}. Value "
+                f"provided was '{next_step}'"
+            )
+            raise ValueError(message)
+
+        # Valid run type
+        run_type = input_data["run-type"]
+        if run_type not in cls.VALID_RUN_TYPES:
+            message = (
+                f"Input 'run-type' value must be one of: {cls.VALID_RUN_TYPES}. Value "
+                f"provided was '{run_type}'"
+            )
+            raise ValueError(message)
+
+        # If next step is extract step, required harvest fields are present
+        # ruff: noqa: SIM102
+        if input_data["next-step"] == "extract":
+            if input_data["source"] not in cls.GIS_SOURCES:
+                if missing_harvest_fields := [
+                    field
+                    for field in cls.REQUIRED_OAI_HARVEST_FIELDS
+                    if field not in input_data
+                ]:
+                    message = (
+                        "Input must include all required harvest fields when starting "
+                        f"with harvest step. Missing fields: {missing_harvest_fields}"
+                    )
+                    raise ValueError(message)
```

Comment:
Hmm, what do you think about moving `validate_input` into [`lambdas/helpers.py`](https://github.com/MITLibraries/timdex-pipeline-lambdas/blob/TIMX-521-new-etl-records-data-location/lambdas/helpers.py)? 🤔

Reply:
I dig it!  I stuck it in the `Config` class as it was previously in the `config.py` file, but if we're going to start moving things around, might as well be where it ultimately feels right.  And agreed, it's `helpers.py`.